### PR TITLE
Edit profile form

### PR DESCRIPTION
**Pull request by julesfll**
# What I did
Still working on it
## Issues addressed

# Screenshots

### CODE REVIEW

**Comment by j-alicia-long** [comment]
@julesfll Make sure you're running ./precommit to check tests & pylint before you push changes so your builds won't fail on Travis :)

**Comment by julesfll** [comment]
# Status
Done

# What I did
Remade profile page and added an edit form:
- Removed student id and school
- Replaced with pre-populated form
- Form includes first name, last name, and year
- Will add major later

## Issues addressed
Lack of edit functionality in form

(editing name and year currently has no effect on other features, it is just for display)

## Screenshots
image

File: tcf_website/views/profile.py [hunk 1]
```diff
@@ -2,12 +2,50 @@
...
+from ..models import User
+
+
+class ProfileForm(ModelForm):
+    """Form updating user profile."""
+    class Meta:
+        model = User
+        fields = ['first_name', 'last_name', 'graduation_year']
+
+        # Add the form-control class to make the form work with Bootstrap
```
Comment (j-alicia-long) [inline tcf_website/views/profile.py:18]: awesome

File: tcf_website/views/profile.py [hunk 2]
```diff
@@ -2,12 +2,50 @@
...
-    return render(request, 'profile/profile.html')
+    if request.method == 'POST':
+        form = ProfileForm(
+            request.POST,
+            label_suffix='',
+            instance=request.user)
+
+        if form.is_valid():
+            form.save()
+            messages.success(request, 'Your profile was updated succesfully!')
```
Comment (j-alicia-long) [inline tcf_website/views/profile.py:42]: oof fix that typo there

File: tcf_core/settings.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 """
 
 import os
+from django.contrib.messages import constants as messages
```
Comment (j-alicia-long) [inline tcf_core/settings.py:15]: ay this is pretty cool stuff

File: tcf_website/templates/base/messages.html [hunk 4]
```diff
@@ -7,7 +7,7 @@
 {% if messages %}
     <div class="content">
         {% for message in messages %}
-            <div class="alert alert-{{message.tags}} alert-dismissible fade show">
+            <div class="alert {{message.tags}} alert-dismissible fade show">
```
Comment (j-alicia-long) [inline tcf_website/templates/base/messages.html:10]: nice modularity

so we can use this for alerts all over the site then yay

**Review by j-alicia-long — COMMENTED** [review]
Looks good! Just a quick typo on "successfully" I noticed (might've been from before)

Also I mentioned before that having no other images on the site makes this one look a little out of place, so maybe we'll have to use that collage somewhere hmm

**Comment by alex-shen1** [comment]
Any idea why "Files changed" has several random "empty" files with 0 lines changed listed? Curious about that.

File: tcf_website/views/profile.py (hunk 2 above)
Comment (julesfll) [reply tcf_website/views/profile.py:42]:
```suggestion
            messages.success(request, 'Your profile was updated successfully!')
```

**Comment by julesfll** [comment]
> Any idea why "Files changed" has several random "empty" files with 0 lines changed listed? Curious about that.

Yeah I have no idea, I thought it could be a line ending issue but it doesn't seem to be

**Review by j-alicia-long — APPROVED** [review]
I think it's good! Once this gets merged we can modify the form on the collect majors PR

**Pull request merged**